### ISSUE TITLE
[HOLD for payment 2023-03-13] [$4000] Able to change timezone through direct link even when the automatically determine location switch is on

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14340]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to Settings > Profile
2. Go to Timezone
3. Toggle off the automatically determine location switch
4. Click on Timezone
5. Copy the URL of the select timezone page
6. Go back and toggle on the automatically determine location switch
7. In the address bar, paste the copied URL and press enter

## Expected Result:
The controls should be disabled, with only the ability to go back

## Actual Result:
After updating the timezone directly through the link, the new timezone is displayed on the timezone initial page, but
The old timezone is displayed on the profile page
The old timezone is selected on the timezone list page
The automatically determine location switch doesn't work

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] MacOS / Chrome / Safari

**Version Number:** 1.2.54-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/212775542-df72ccb2-295f-42d5-9ef7-fe337d75ce76.mov

https://user-images.githubusercontent.com/43996225/212775561-8049cef7-55d7-4302-86ad-56688a900e94.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @adeel0202 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673873060216069

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b8d0920a25d93b4f</li>
        <li>Upwork Job ID: 1618711874173829120</li>
        <li>Last Price Increase: 2023-02-09</li>
    </ul>
</details>

**Comment by sonialiap** [comment]
I can replicate the reported behavior. Setting the timezone via the link does not update the timezone set on my profile or in the timezone selection list. Looks like making the change via the link does not override the "automatically determine toggle"

**Comment by sonialiap** [comment]
Asking team about expected behavior [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1674519465420829)

**Comment by sonialiap** [comment]
Team agreed that the behavior described here isn't ideal. Suggested solution is when `Automatically determine your location` then redirect https://new.expensify.com/settings/profile/timezone/select back to https://new.expensify.com/settings/profile/timezone

**Comment by Pujan92** [comment]
**Proposal**

Check the timezone `automatic` value and based on that navigate to initial page from constructor.

```diff
iff --git a/src/pages/settings/Profile/TimezoneSelectPage.js b/src/pages/settings/Profile/TimezoneSelectPage.js
index 7a89ac8ce7..fe31c60405 100644
--- a/src/pages/settings/Profile/TimezoneSelectPage.js
+++ b/src/pages/settings/Profile/TimezoneSelectPage.js
@@ -34,7 +34,13 @@ class TimezoneSelectPage extends Component {
         this.saveSelectedTimezone = this.saveSelectedTimezone.bind(this);
         this.filterShownTimezones = this.filterShownTimezones.bind(this);
 
-        this.currentSelectedTimezone = lodashGet(props.currentUserPersonalDetails, 'timezone.selected', CONST.DEFAULT_TIME_ZONE.selected);
+        const timezoneDetails = lodashGet(props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
+        this.currentSelectedTimezone = timezoneDetails.selected;
+
+        if (timezoneDetails.automatic) {
+            Navigation.navigate(ROUTES.SETTINGS_TIMEZONE);
+        }
+
         this.allTimezones = _.chain(moment.tz.names())
```

**Comment by bernhardoj** [comment]
### **Proposal**

I think we need to replace the route so the user can't press back and land on the timezone select page.

First, we need to introduce a new parameter to the navigate function whether we want to replace the route or not. Then, just like the same thing the others done above is to check whether the timezone is set to automatic or not. If yes, we will navigate to timezone initial page with the replace set to true.

```diff
diff --git a/src/libs/Navigation/Navigation.js b/src/libs/Navigation/Navigation.js
index 3635a7840..f2d38e207 100644
--- a/src/libs/Navigation/Navigation.js
+++ b/src/libs/Navigation/Navigation.js
@@ -132,7 +132,7 @@ function isDrawerRoute(route) {
  * Main navigation method for redirecting to a route.
  * @param {String} route
  */
-function navigate(route = ROUTES.HOME) {
+function navigate(route = ROUTES.HOME, replace = false) {
     if (!canNavigate('navigate', {route})) {
         // Store intended route if the navigator is not yet available,
         // we will try again after the NavigationContainer is ready
@@ -159,7 +159,7 @@ function navigate(route = ROUTES.HOME) {
         return;
     }
 
-    linkTo(navigationRef.current, route);
+    linkTo(navigationRef.current, route, replace);
 }
 
 /**
diff --git a/src/libs/Navigation/linkTo.js b/src/libs/Navigation/linkTo.js
index 04977e6a6..3c6af3a2a 100644
--- a/src/libs/Navigation/linkTo.js
+++ b/src/libs/Navigation/linkTo.js
@@ -1,10 +1,12 @@
 import {
     getStateFromPath,
     getActionFromState,
+    StackActions,
 } from '@react-navigation/core';
 import linkingConfig from './linkingConfig';
 
-export default function linkTo(navigation, path) {
+export default function linkTo(navigation, path, replace) {
     const normalizedPath = !path.startsWith('/') ? `/${path}` : path;
     if (navigation === undefined) {
         throw new Error("Couldn't find a navigation object. Is your component inside a screen in a navigator?");
@@ -27,7 +29,11 @@ export default function linkTo(navigation, path) {
         root = current;
     }
 
-    const action = getActionFromState(state, linkingConfig.config);
+    let action = getActionFromState(state, linkingConfig.config);
+
+    if (replace) {
+        action = {...action, type: 'REPLACE'};
+    }
 
     if (action !== undefined) {
         root.dispatch(action);
diff --git a/src/pages/settings/Profile/TimezoneSelectPage.js b/src/pages/settings/Profile/TimezoneSelectPage.js
index 7a89ac8ce..28b6b214e 100644
--- a/src/pages/settings/Profile/TimezoneSelectPage.js
+++ b/src/pages/settings/Profile/TimezoneSelectPage.js
@@ -34,7 +34,11 @@ class TimezoneSelectPage extends Component {
         this.saveSelectedTimezone = this.saveSelectedTimezone.bind(this);
         this.filterShownTimezones = this.filterShownTimezones.bind(this);
 
-        this.currentSelectedTimezone = lodashGet(props.currentUserPersonalDetails, 'timezone.selected', CONST.DEFAULT_TIME_ZONE.selected);
+        this.currentTimezone = lodashGet(props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
+        this.currentSelectedTimezone = this.currentTimezone.selected;
+        if (this.currentTimezone.automatic) {
+            Navigation.navigate(ROUTES.SETTINGS_TIMEZONE, true);
+        }
         this.allTimezones = _.chain(moment.tz.names())
             .filter(timezone => !timezone.startsWith('Etc/GMT'))
             .map(timezone => ({
@@ -82,7 +86,7 @@ class TimezoneSelectPage extends Component {
                     onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_TIMEZONE)}
                     onCloseButtonPress={() => Navigation.dismissModal(true)}
                 />
-                <OptionsSelector
+                {!this.currentTimezone.automatic && <OptionsSelector
                     textInputLabel={this.props.translate('timezonePage.timezone')}
                     value={this.state.timezoneInputText}
                     onChangeText={this.filterShownTimezones}
@@ -90,7 +94,7 @@ class TimezoneSelectPage extends Component {
                     optionHoveredStyle={styles.hoveredComponentBG}
                     sections={[{data: this.state.timezoneOptions}]}
                     shouldHaveOptionSeparator
-                />
+                />}
             </ScreenWrapper>
         );
     }

```

I also add !this.currentTimezone.automatic to not render OptionsSelector to prevent keyboard blinking as shown from the video below.

https://user-images.githubusercontent.com/50919443/215002924-6983d2ac-b819-4402-8a52-92546e8319b4.mp4


After applying the condition, here is how it looks.


https://user-images.githubusercontent.com/50919443/215002936-0d6d97f1-925e-405e-afec-9c2869e51905.mp4

**Comment by s77rt** [comment]
### PreProposal
<details>
<summary>The TimezoneSelectPage component suffers from stale data issue and proposals won't work as intended.</summary>

The stale data is on `currentUserPersonalDetails` where if the timezone get's updated whether to be set to another timezone or switched to automatic, the page won't re-render.

Although technically this is a different issue it seems necessary to fix the reported bug. This is how I'd fix the stale data issue:

```diff
diff --git a/src/pages/settings/Profile/TimezoneSelectPage.js b/src/pages/settings/Profile/TimezoneSelectPage.js
index 7a89ac8ce7..634e947be9 100644
--- a/src/pages/settings/Profile/TimezoneSelectPage.js
+++ b/src/pages/settings/Profile/TimezoneSelectPage.js
@@ -31,28 +31,51 @@ class TimezoneSelectPage extends Component {
     constructor(props) {
         super(props);
 
+        this.getCurrentTimezone = this.getCurrentTimezone.bind(this);
+        this.getAllTimezones = this.getAllTimezones.bind(this);
         this.saveSelectedTimezone = this.saveSelectedTimezone.bind(this);
         this.filterShownTimezones = this.filterShownTimezones.bind(this);
 
-        this.currentSelectedTimezone = lodashGet(props.currentUserPersonalDetails, 'timezone.selected', CONST.DEFAULT_TIME_ZONE.selected);
-        this.allTimezones = _.chain(moment.tz.names())
+        this.timezone = this.getCurrentTimezone();
+        this.allTimezones = this.getAllTimezones();
+
+        this.state = {
+            timezoneInputText: this.timezone.selected,
+            timezoneOptions: this.allTimezones,
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (lodashGet(prevProps.currentUserPersonalDetails, 'timezone') === lodashGet(this.props.currentUserPersonalDetails, 'timezone')) {
+            return;
+        }
+
+        this.timezone = this.getCurrentTimezone();
+        this.allTimezones = this.getAllTimezones();
+        this.setState({
+            timezoneInputText: this.timezone.selected,
+            timezoneOptions: this.allTimezones,
+        });
+    }
+
+    getCurrentTimezone() {
+        return lodashGet(this.props.currentUserPersonalDetails, 'timezone', CONST.DEFAULT_TIME_ZONE);
+    }
+
+    getAllTimezones() {
+        return _.chain(moment.tz.names())
             .filter(timezone => !timezone.startsWith('Etc/GMT'))
             .map(timezone => ({
                 text: timezone,
                 keyForList: timezone,
 
                 // Include the green checkmark icon to indicate the currently selected value
-                customIcon: timezone === this.currentSelectedTimezone ? greenCheckmark : undefined,
+                customIcon: timezone === this.timezone.selected ? greenCheckmark : undefined,
 
                 // This property will make the currently selected value have bold text
-                boldStyle: timezone === this.currentSelectedTimezone,
+                boldStyle: timezone === this.timezone.selected,
             }))
             .value();
-
-        this.state = {
-            timezoneInputText: this.currentSelectedTimezone,
-            timezoneOptions: this.allTimezones,
-        };
     }
 
     /**
```

**IMPORTANT**: The proposals below are based on that proposal being applied first.

**NOTE**: There is also another issue with `OptionList` where it won't re-render if the data roughly changed since we do use shallow object comparison. The fix is to pass extraData but this will also requires changes to forward the same prop from `OptionsSelector`. This issue is not much noticeable because most of the time the `TimezoneSelectPage` component unmounts then mounts again and the flatlist will render as if it's the first time. Let's not handle this here, just wanted to mention it for now.

</details>


___

### Proposal 1
```diff
diff --git a/src/pages/settings/Profile/TimezoneSelectPage.js b/src/pages/settings/Profile/TimezoneSelectPage.js
index 634e947be9..0fe9581ed3 100644
--- a/src/pages/settings/Profile/TimezoneSelectPage.js
+++ b/src/pages/settings/Profile/TimezoneSelectPage.js
@@ -37,6 +37,9 @@ class TimezoneSelectPage extends Component {
         this.filterShownTimezones = this.filterShownTimezones.bind(this);
 
         this.timezone = this.getCurrentTimezone();
+        if (this.timezone.automatic) {
+            Navigation.navigate(ROUTES.SETTINGS_TIMEZONE);
+        }
         this.allTimezones = this.getAllTimezones();
 
         this.state = {
@@ -51,6 +54,9 @@ class TimezoneSelectPage extends Component {
         }
 
         this.timezone = this.getCurrentTimezone();
+        if (this.timezone.automatic) {
+            Navigation.navigate(ROUTES.SETTINGS_TIMEZONE);
+        }
         this.allTimezones = this.getAllTimezones();
         this.setState({
             timezoneInputText: this.timezone.selected,
```

### Details
After applying the PreProposal we can just follow it with any of the above proposals

However it won't cover all the cases. One case that I have in mind is cache related where you will still get the timezones list. You can experience this issue by following those steps:
1. Navigate to http://localhost:8080/settings/profile/timezone
2. Switch off automatic timezone
3. Open timezone selection list
4. Go back using browser back button
5. Switch on automatic timezone
6. Go forward using browser forward button
7. If necessary go back again using browser back button
8. You will see the timezones list again


https://user-images.githubusercontent.com/16493223/215272245-b5d36525-d4e9-4cd3-aaef-fc556fac2a7b.mp4

___

### Proposal 2
```diff
diff --git a/src/libs/actions/PersonalDetails.js b/src/libs/actions/PersonalDetails.js
index 9f0247a75f..5fadb808d5 100644
--- a/src/libs/actions/PersonalDetails.js
+++ b/src/libs/actions/PersonalDetails.js
@@ -198,8 +198,7 @@ function updateAutomaticTimezone(timezone) {
 }
 
 /**
- * Updates user's 'selected' timezone, then navigates to the
- * initial Timezone page.
+ * Updates user's 'selected' timezone.
  *
  * @param {String} selectedTimezone
  */
@@ -220,7 +219,6 @@ function updateSelectedTimezone(selectedTimezone) {
             },
         }],
     });
-    Navigation.navigate(ROUTES.SETTINGS_TIMEZONE);
 }
 
 /**
diff --git a/src/pages/settings/Profile/TimezoneSelectPage.js b/src/pages/settings/Profile/TimezoneSelectPage.js
index 634e947be9..9e135a4321 100644
--- a/src/pages/settings/Profile/TimezoneSelectPage.js
+++ b/src/pages/settings/Profile/TimezoneSelectPage.js
@@ -83,7 +83,15 @@ class TimezoneSelectPage extends Component {
      * @param {String} timezone.text
      */
     saveSelectedTimezone({text}) {
-        PersonalDetails.updateSelectedTimezone(text);
+        if (this.timezone.automatic) {
+            PersonalDetails.updateAutomaticTimezone({
+                automatic: false,
+                selected: text,
+            })
+        } else {
+            PersonalDetails.updateSelectedTimezone(text);
+        }
+        Navigation.navigate(ROUTES.SETTINGS_TIMEZONE);
     }
 
     /**
```

### Details
This proposals is based on the initially mentioned expected results.


https://user-images.githubusercontent.com/16493223/215272583-3290a94d-df35-405a-bb47-5491db24f9bb.mp4


___

### Proposal 1 vs. Proposal 2

Personally, I think we should go with proposal 2 for two reasons: 
1. It is the simplest solution to implement
2. If the user navigates to that URL, he is implicitly asking to set his timezone manually. Thus we should not redirect him.

**Comment by mollfpr** [comment]
Thank you for the proposals!

@Pujan92 @bernhardoj Your proposal is working fine, but it's only on the first try. After redirecting to the settings timezone page and pressing the go back button, then opening the direct link again, it will show the timezone select page.

@bernhardoj Thanks for noticing the keyboard visibility while redirecting to the settings timezone page! For your first proposal, I don't think we want to change anything in `Navigation`; it's currently on refactoring.

Also, could you explain why putting the navigation inside the _constructor_ rather than in the `componentDidMount`?

For the keyboard appearance issue, we can automatically set the `OptionsSelector` auto-focus based on the timezone.

**Comment by mollfpr** [comment]
I'll reviewing @s77rt and @fedirjh proposal's some how I'm miss the proposals.

**Comment by Pujan92** [comment]
> Also, could you explain why putting the navigation inside the _constructor_ rather than in the `componentDidMount`?

It is to redirect user asap with skipping extra logic execution. But yes it will work only first time.

**Comment by bernhardoj** [comment]
I agree with setting the auto focus instead.

Does my second proposal works well or suffer the same issue with other proposals? 

> After redirecting to the settings timezone page and pressing the go back button, then opening the direct link again, it will show the timezone select page.

I can't seem to reproduce this. Every time we open the direct link again, the web is refreshed and I believe 
constructor/componentDidMount will be called again.

> Also, could you explain why putting the navigation inside the constructor rather than in the componentDidMount?

I have the same thought with @Pujan92.

**Comment by mollfpr** [comment]
@bernhardoj Your second proposal is not working with browser back button.

https://user-images.githubusercontent.com/25520267/216094689-adfa3fce-c520-4faa-a1bf-3ef87a8fdb76.mov

---
>Suggested solution is when Automatically determine your location then redirect https://new.expensify.com/settings/profile/timezone/select back to https://new.expensify.com/settings/profile/timezone

@tylerkaraszewski I think the above suggested solution is will not work perfectly and we don't have good proposal for that. I'll say we keep the follow the OP expected result like on @s77rt and @fedirjh proposals?

**Comment by mollfpr** [comment]
>@mollfpr trying to follow this issue here. Can you elaborate on what you mean by following the expected results in the two different proposals?

@kevinksullivan  Currently, the OP expected result in the description and the suggested solution https://github.com/Expensify/App/issues/14340#issuecomment-1403229409 is different. While reviewing the proposal that follows the suggested solution, there's a similar problem regarding the navigation. We can get around the problem, but that will look like a hack.

I think we should follow the OP's expected result in the description. The example is on the @fedirjh [proposal](https://github.com/Expensify/App/issues/14340#issuecomment-1407910719), where it makes sense to me to let the user changes the timezone when opening the direct link to the select timezone page.

**Comment by MelvinBot** [comment]
@tylerkaraszewski @kevinksullivan @mollfpr this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by MelvinBot** [comment]
@tylerkaraszewski @kevinksullivan @mollfpr this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by MelvinBot** [comment]
@tylerkaraszewski @kevinksullivan @mollfpr this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by kevinksullivan** [comment]
Discussing this one here

https://expensify.slack.com/archives/C01GTK53T8Q/p1675722834448009

**Comment by priyeshshah11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
User should not be able to change timezone through direct link when the automatically determine location switch is on 

### What is the root cause of that problem?
We don't have check for disabling the list options on the timezone select page.

### What changes do you think we should make in order to solve the problem?
We should disable the list by passing the `isDisabled` prop as true on the TimeZoneSelectPage when automatic timezone is selected or turned on

**Comment by tylerkaraszewski** [comment]
@mollfpr do you have a preferred proposal here?

**Comment by mollfpr** [comment]
@tylerkaraszewski Ask for clarity at the [Slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1676041227774079?thread_ts=1675722834.448009&cid=C01GTK53T8Q) before continue to review the proposal.

**Comment by kevinksullivan** [comment]
Looks like we're good to keep moving forward here. We are nearing the 4 week timeline though. @mollfpr I updated the expected behavior in the OP. I don't think we had any proposals aligned with this approach, but perhaps we can source them quickly from those that were working on the other approaches?

**Comment by mollfpr** [comment]
@kevinksullivan @tylerkaraszewski Based on our new expected result @priyeshshah11 [proposal](https://github.com/Expensify/App/issues/14340#issuecomment-1424471308) looks like what we looking for.

**Comment by farolanf** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
User is able to change timezone through direct link even if timezone is set to automatic.

### What is the root cause of that problem?
Not handling the case when timezone is set to automatic on the timezone select page.

### What changes do you think we should make in order to solve the problem?
1. Disable selection if timezone is set to automatic.
2. Show a message to inform the user why timezone can't be changed.

#### Example screen of disabled selection and the message.
![New Expensify 2023-02-13 15-38-26](https://user-images.githubusercontent.com/15892844/218399215-f73652b9-c259-406a-97c0-fdc034640471.png)

### What alternative solutions did you explore? (Optional)

**Comment by priyeshshah11** [comment]
@mollfpr @tylerkaraszewski @kevinksullivan just waiting for the issue to be assigned to raise a PR.

**Comment by mdneyazahmad** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Timezone is updatable via link even if this is set to automatic. 

### What is the root cause of that problem?

When timezone is set to automatic, we are disabling the button which when clicked will navigate to  timezone selection screen. The issue here is that the user can navigate to that screen via links, and this is not protected.

### What changes do you think we should make in order to solve the problem?


There are already proposal inclined towards the agreed expected behaviour. I am proposing a slightly different behaviour.

If the user navigates to that page via link (very rare), and selects a timezone, `PersonalDetails.updateSelectedTimezone` is executed. 

https://github.com/Expensify/App/blob/8dffdf8a4ce1685b0b5b8cecbfa3938948e3dfcc/src/libs/actions/PersonalDetails.js#L278

We will prevent this api call if the timezone is automatic, and show user a message that he can not update it.

```
function updateSelectedTimezone(isTimezoneAutomatic, selectedTimezone) {
    if (isTimezoneAutomatic) {
        Navigation.navigate(ROUTES.SETTINGS_TIMEZONE);
        Growl.error('You can not update timezone when this is set to automatic.');

        return;
    }
```

We can pass `isTimezoneAutomatic` from `TimezoneSelectPage` or hook into onyx in `PersonalDetails`.

```
// TimezoneSelectPage.js
    saveSelectedTimezone({text}) {
        const isTimezoneAutomatic = lodashGet(this.props.currentUserPersonalDetails, 'timezone.automatic', false);
        PersonalDetails.updateSelectedTimezone(isTimezoneAutomatic, text);
    }

```

I see that disabling things is less preferred in this project, so this solution may be inclined towards our philosophy.

*Option*: I would suggest to protect this api call on the backed side. The server should throw an error when update timezone is performed and this is set to automatic.

### What alternative solutions did you explore? (Optional)
NA

**Comment by MelvinBot** [comment]
@tylerkaraszewski @kevinksullivan @mollfpr this issue is now 4 weeks old and preventing us from maintaining WAQ, can you:
- Decide whether any proposals currently meet our guidelines and can be approved as-is today
- If no proposals meet that standard, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #expensify-open-source

Thanks!

**Comment by MelvinBot** [comment]
Current assignee @mollfpr is eligible for the Internal assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
@tylerkaraszewski @kevinksullivan @mollfpr this issue is now 4 weeks old and preventing us from maintaining WAQ. This should now be your highest priority. Please post below what your plan is to get a PR in review ASAP. Thanks!

**Comment by mollfpr** [comment]
Bump @tylerkaraszewski @kevinksullivan

**Comment by MelvinBot** [comment]
Current assignee @kevinksullivan is eligible for the External assigner, not assigning anyone new.

**Comment by kevinksullivan** [comment]
LEt's move ahead with @priyeshshah11 's proposal. I'm sending an offer in upwork now.

**Comment by priyeshshah11** [comment]
> LEt's move ahead with @priyeshshah11 's proposal. I'm sending an offer in upwork now.

Thanks @kevinksullivan, accepted on upwork. Could you please assign me the issue as well? 😃

**Comment by mdneyazahmad** [comment]
@mollfpr can you please review my proposal https://github.com/Expensify/App/issues/14340#issuecomment-1428194189. I will appreciate it if you give some comment about it. Thanks!

**Comment by mdneyazahmad** [comment]
@mollfpr can you please review my proposal https://github.com/Expensify/App/issues/14340#issuecomment-1428194189. I will appreciate it if you give some comment about it. Thanks!

**Comment by priyeshshah11** [comment]
@mollfpr PR is up here https://github.com/Expensify/App/pull/15176

**Comment by mollfpr** [comment]
@farolanf @mdneyazahmad Thank you for the proposals!

@farolanf The difference between your proposal and the selected proposal is that you added an error message. I'm unsure about adding the error message because the user hasn't taken any action on the page. If you think this is necessary you can start a discussion about it and request to open an issue for your suggestion.

@mdneyazahmad I'm more inclined to prevent it before the user takes action or let the user can update the timezone because the backend allows it. But we already decide to continue disabling the option selection.

**Comment by CortneyOfstad** [comment]
Assigning @priyeshshah11 per Slack convo [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1676991224967129).

**Comment by MelvinBot** [comment]
@tylerkaraszewski, @kevinksullivan, @mollfpr, @priyeshshah11 Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@tylerkaraszewski, @kevinksullivan, @mollfpr, @priyeshshah11 Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.78-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15176


If no regressions arise, payment will be issued on 2023-03-13. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mollfpr / @tylerkaraszewski] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mollfpr / @tylerkaraszewski] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mollfpr / @tylerkaraszewski] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@priyeshshah11] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@kevinksullivan] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by tylerkaraszewski** [comment]
@mollfpr - can you identify the PR that introduced this?

**Comment by mollfpr** [comment]
@tylerkaraszewski Sorry for the delay. I wonder if any of the PR is causing this bug. Because the expected result is to disable the selection depending on the data, it's something new added in the PR.

**Comment by priyeshshah11** [comment]
Regression Test Proposal

1. Go to Settings -> Profile -> Timezone.
2. Toggle ON automatic timezone. 
3. Go to https://new.expensify.com/settings/profile/timezone/select in the browser.
4. Verify that items in the list are disabled.

Do we agree 👍 or 👎

cc: @mollfpr @tylerkaraszewski

**Comment by kevinksullivan** [comment]
@adeel0202 @mollfpr sent you separate offers since the upwork job was closed for new applicants.

**Comment by mollfpr** [comment]
@kevinksullivan Accepted with the speed of light ⚡️

**Comment by kevinksullivan** [comment]
Paid everyone. @tylerkaraszewski let us know what you think [here](https://github.com/Expensify/App/issues/14340#issuecomment-1471873763) and then we can close this out.

**Comment by priyeshshah11** [comment]
Thanks for the payment @kevinksullivan, just wondering whether this qualifies for the bonus? As all the work by me & the review from @mollfpr was done within 3 business days, we were just waiting for @tylerkaraszewski's review for quite some time.

**Comment by situchan** [comment]
12k total (including C+ review and timeline bonus) is huge bounty for this issue 😮 
Price keep doubled while discussing the expected behavior, and the final solution is not that complex.

**Comment by Pujan92** [comment]
@priyeshshah11 @tylerkaraszewski @mollfpr @kevinksullivan  Linked [PR](https://github.com/Expensify/App/pull/15176) to this issue broke the scrolling to initial focused option functionality. 

https://user-images.githubusercontent.com/14358475/227557034-23ed7863-afc0-45b9-aa5f-f6cfda96e3f1.mov

------
**Sharing possible solution**

1. https://github.com/Expensify/App/blob/10d3d52e993303db477478d323af925f4d439acf/src/pages/settings/Profile/TimezoneSelectPage.js#L142 
`this.currentSelectedTimezone` should be replaced with `this.timezone.selected`

2. https://github.com/Expensify/App/blob/10d3d52e993303db477478d323af925f4d439acf/src/pages/settings/Profile/TimezoneSelectPage.js#L74-L76
I don't know why this(`getKey`) is required but if it remains then we need to consider `text` here instead of `keyForList`
https://github.com/Expensify/App/blob/10d3d52e993303db477478d323af925f4d439acf/src/components/OptionsSelector/BaseOptionsSelector.js#L159

https://user-images.githubusercontent.com/14358475/227557172-44e48137-281c-4dca-a52e-eb6c719d6915.mov

**Comment by priyeshshah11** [comment]
Is there any action required by me?

**Comment by kevinksullivan** [comment]
Was just checking our policy on regressions beyond 7 days. It looks like this should still qualify for the bonus. Sent another offer to get this squared away @priyeshshah11 @mollfpr 

https://www.upwork.com/jobs/~01295b5353bdb74897

**Comment by priyeshshah11** [comment]
> Was just checking our policy on regressions beyond 7 days. It looks like this should still qualify for the bonus. Sent another offer to get this squared away @priyeshshah11 @mollfpr 
> 
> 
> 
> https://www.upwork.com/jobs/~01295b5353bdb74897

Thanks @kevinksullivan but the offers shows as withdrawn for me and I can't accept/apply. Could you please resend?

**Comment by mollfpr** [comment]
Thanks @kevinksullivan accepted!

**Comment by kevinksullivan** [comment]
@priyeshshah11 hm, upwork seems to be throwing an error each time I try to invite you. I'll try again.

**Comment by kevinksullivan** [comment]
I don't know what's going on here. It makes it seem as though it worked and then doesn't show you in the list. Want to try applying and I'll accept the proposal from there? Sorry about this @priyeshshah11

**Comment by priyeshshah11** [comment]
@kevinksullivan Got the offer and accepted now! Thanks 👍

**Comment by MelvinBot** [comment]
@tylerkaraszewski, @kevinksullivan, @mollfpr, @priyeshshah11 Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by tylerkaraszewski** [comment]
@kevinksullivan - this issue shows as overdue but I'm not sure what action we're waiting on. A PR from @priyeshshah11 ?

**Comment by kevinksullivan** [comment]
Just waiting to finish payment @tylerkaraszewski

**Comment by kevinksullivan** [comment]
@priyeshshah11 it looks like the offer is still pending... Does it show as accepted on your end?

<img width="641" alt="image" src="https://user-images.githubusercontent.com/22508290/229959737-a402d064-fa44-4890-9747-9cbb1b88ab8a.png">

**Comment by priyeshshah11** [comment]
> @priyeshshah11 it looks like the offer is still pending... Does it show as accepted on your end?
> 
> 
> 
> <img width="641" alt="image" src="https://user-images.githubusercontent.com/22508290/229959737-a402d064-fa44-4890-9747-9cbb1b88ab8a.png">
> 
> 

@kevinksullivan accepted & requested payment 

![image](https://user-images.githubusercontent.com/38547776/230010287-cbaacbe9-6f5e-4e68-ab40-461ed6ce7d50.png)

**Comment by kevinksullivan** [comment]
![image](https://user-images.githubusercontent.com/22508290/230624556-de249ea4-7f54-4c50-9ac4-814b3a005717.png)


sheesh, sorry about all the back and forth @priyeshshah11 . It seems I finally got this to work. Closing out!
